pkg/authn/k8schain/k8schain.go

```diff
@@ -42,60 +46,35 @@ type Options struct {
 	ImagePullSecrets []string
 }
 
-var (
-	keyring credentialprovider.DockerKeyring
-	once    sync.Once
-)
-
 // New returns a new authn.Keychain suitable for resolving image references as
 // scoped by the provided Options.  It speaks to Kubernetes through the provided
 // client interface.
+//
+// Deprecated: Use pkg/authn/kubernetes.
```

Comment:
It is probably worth separating the conversation about Deprecation from this PR, because I don't see the value in forcing all of the folks that are using `k8schain` to drop it so that we can stop maintaining what is now a very small shim.  We can always add these later if we can agree on a path forward for folks.

To me the main win here is cutting out the need for Vincent's fork, which is HUGE, and I totally appreciate all of the work (and arguing!) to get us to this point.  When the world stops ending, I owe you some beverages 🍻

Reply:
Yeah, you're probably right. Done.